packages/fxa-settings/src/models/integrations/data/data.ts

```diff
@@ -0,0 +1,357 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {
+  IsBase64,
+  IsBoolean,
+  IsEmail,
+  IsHexadecimal,
+  IsIn,
+  IsInt,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  Length,
+  MaxLength,
+  MinLength,
+  Validate,
+} from 'class-validator';
+import {
+  bind,
+  KeyTransforms as T,
+  ModelDataProvider,
+} from '../../../lib/model-data';
+import { IsFxaRedirectToUrl, IsFxaRedirectUri } from '../../../lib/validation';
+
+/**
+ * Base integration class. Fields in this class represents data commonly accessed across many pages and is useful for various flows.
+ */
+export class IntegrationData extends ModelDataProvider {
+  // TODO: Should this only exist on OAuthIntegrationData
+  @IsOptional()
+  @IsString()
+  @bind()
+  public clientId: string | undefined;
+
+  @IsOptional()
+  @IsString()
+  @bind()
+  public service: string | undefined;
+
+  // TODO - Validation - This should be a URL, but it is encoded and must be decoded in order to validate.
+  @IsOptional()
+  @Validate(IsFxaRedirectToUrl, {})
+  @bind(T.snakeCase)
+  redirectTo: string | undefined;
+
+  // TODO: Double check context. This might be sync data!
```

Comment:
Context is required for sync (at least, for `oauth_desktop_v1`). If signing in to sync and `context` is omitted, I see this popup:

![Screenshot 2025-04-15 at 12 11 52 PM](https://github.com/user-attachments/assets/2fa54bf3-58c8-4792-a663-66a770014885)

Reply:
Should we add this logic to query param model?

Reply:
Not sure what to do here. I'm guessing sync always provides context anyways, but regardless, this seems like something to check for and require on the sign in page.